### PR TITLE
[Gardening] Fix typos across docs and codebase

### PR DESCRIPTION
**Pull request by L-j-h-c**
sucessfully -> successfully
acknowledgement -> acknowledgment
prefered -> preferred
refered -> referred
completly -> completely
occurences -> occurrences
occured -> occurred
occuring -> occurring
publically -> publicly
Therefor -> Therefore
agressively -> aggressively
immediatly -> immediately
enviroment -> environment
accross -> across
seperator -> separator

### CODE REVIEW

**Comment by L-j-h-c** [comment]
@swift-ci please smoke test

File: docs/DifferentiableProgramming.md [hunk 1]
```diff
@@ -108,7 +108,7 @@ Backticks were added manually.
     *   [Not support differentiable programming](#not-support-differentiable-programming)
     *   [Use another language or framework for differentiable programming](#use-another-language-or-framework-for-differentiable-programming)
     *   [Other approaches to differentiable programming](#other-approaches-to-differentiable-programming)
-*   [Acknowledgements](#acknowledgements)
+*   [Acknowledgments](#acknowledgments)
```
Comment (AnthonyLatsis) [inline docs/DifferentiableProgramming.md:111]: This is the British spelling, not a typo.

File: docs/DifferentiableProgrammingImplementation.md [hunk 2]
```diff
@@ -993,8 +993,8 @@ See [here](DifferentiableProgramming.md#linear-maps) for more information about
 - VJP functions will be removed throughout the differentiation system.
 - Pullback function generation will change to use transposition.
 
-## Acknowledgements
+## Acknowledgments
```
Comment (AnthonyLatsis) [inline docs/DifferentiableProgrammingImplementation.md:996]: Ditto.

File: utils/gen-unicode-data/Sources/GenScalarProps/GeneralCategory.swift [hunk 3]
```diff
@@ -35,9 +35,9 @@ enum GeneralCategory: String {
   case currencySymbol = "Sc"
   case modifierSymbol = "Sk"
   case otherSymbol = "So"
-  case spaceSeperator = "Zs"
-  case lineSeperator = "Zl"
-  case paragraphSeperator = "Zp"
+  case spaceseparator = "Zs"
+  case lineseparator = "Zl"
+  case paragraphseparator = "Zp"
```
Comment (AnthonyLatsis) [inline utils/gen-unicode-data/Sources/GenScalarProps/GeneralCategory.swift:40]: Camel case please.

**Review by AnthonyLatsis — COMMENTED** [review]
Thanks. Can you edit the commit message to use the `[Gardening]` tag instead? `[Fix]` is too vague.

File: docs/DifferentiableProgramming.md (hunk 1 above)
Comment (L-j-h-c) [reply docs/DifferentiableProgramming.md]: Thanks, I didn't catch that. I fixed it.

File: utils/gen-unicode-data/Sources/GenScalarProps/GeneralCategory.swift (hunk 3 above)
Comment (L-j-h-c) [reply utils/gen-unicode-data/Sources/GenScalarProps/GeneralCategory.swift]: Thanks, I Applied Camel case.

**Review by AnthonyLatsis — APPROVED** [review]
Looks good! Let us wait for a member to approve and we should be good to go then.

**Review by ahoppen — APPROVED** [review]
Thanks. LGTM

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Review by xedin — APPROVED** [review]
Thank you!

**Pull request merged**